src/main.cpp

```diff
@@ -2293,10 +2293,11 @@ static int64_t nTimeIndex = 0;
 static int64_t nTimeCallbacks = 0;
 static int64_t nTimeTotal = 0;
 
-bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pindex,
+bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* const pindex,
                   CCoinsViewCache& view, const CChainParams& chainparams, bool fJustCheck)
 {
     AssertLockHeld(cs_main);
+    assert(pindex->nVersion == block.nVersion);
```

Comment:
I added the assert, because if that's not the case, then my change is not without side effect. I reviewed the caller, and it seems that pindex always point to block's CBlockIndex.